File: Examples/Android/SignUpFormDemo.kt
// Android Example - Kotlin  
// Comprehensive Sign-Up Form with Full Validation

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewmodel.compose.viewModel

/**
 * Comprehensive Sign-Up Form ViewModel
 * 
 * Demonstrates complex form validation using UniCoreValidation
 * across multiple fields with real-time feedback
 */
class SignUpFormViewModel : ViewModel() {
    // Form fields
    val firstName = mutableStateOf("")
    val lastName = mutableStateOf("")
    val email = mutableStateOf("")
    val username = mutableStateOf("")
    val password = mutableStateOf("")
    val confirmPassword = mutableStateOf("")
    
    // Validation state
    val firstNameError = mutableStateOf<ValidationErrorModel?>(null)
    val lastNameError = mutableStateOf<ValidationErrorModel?>(null)
    val emailError = mutableStateOf<ValidationErrorModel?>(null)
    val usernameError = mutableStateOf<ValidationErrorModel?>(null)
    val passwordError = mutableStateOf<ValidationErrorModel?>(null)
    val confirmPasswordError = mutableStateOf<ValidationErrorModel?>(null)
    
    val isFirstNameValid = mutableStateOf(false)
    val isLastNameValid = mutableStateOf(false)
    val isEmailValid = mutableStateOf(false)
    val isUsernameValid = mutableStateOf(false)
    val isPasswordValid = mutableStateOf(false)
    val isConfirmPasswordValid = mutableStateOf(false)
    
    fun updateFirstName(value: String) {
        firstName.value = value
        validateFirstName()
    }
    
    fun updateLastName(value: String) {
        lastName.value = value
        validateLastName()
    }
    
    fun updateEmail(value: String) {
        email.value = value
        validateEmail()
    }
    
    fun updateUsername(value: String) {
        username.value = value
        validateUsername()
    }
    
    fun updatePassword(value: String) {
        password.value = value
        validatePassword()
    }
    
    fun updateConfirmPassword(value: String) {
        confirmPassword.value = value
        validateConfirmPassword()
    }
    
    private fun validateFirstName() {
        val result = validateTextField(firstName.value, 2, 50)
        isFirstNameValid.value = result.isValid
        firstNameError.value = result.errors?.firstOrNull()
    }
    
    private fun validateLastName() {
        val result = validateTextField(lastName.value, 2, 50)
        isLastNameValid.value = result.isValid
        lastNameError.value = result.errors?.firstOrNull()
    }
    
    private fun validateEmail() {
        val errors = mutableListOf<ValidationErrorModel>()
        
        if (email.value.isEmpty()) {
            errors.add(
                ValidationErrorModel(
                    code = "required",
                    messageKey = "validation.required",
                    metadata = emptyMap()
                )
            )
        }
        
        val emailPattern = "^[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\\.[A-Z|a-z]{2,}$"
        if (!Regex(emailPattern).matches(email.value) && email.value.isNotEmpty()) {
            errors.add(
                ValidationErrorModel(
                    code = "email.invalid",
                    messageKey = "validation.email.invalid",
                    metadata = emptyMap()
                )
            )
        }
        
        val result = if (errors.isEmpty()) {
            ValidationResultModel(isValid = true, errors = null)
        } else {
            ValidationResultModel(isValid = false, errors = errors)
        }
        
        isEmailValid.value = result.isValid
        emailError.value = result.errors?.firstOrNull()
    }
    
    private fun validateUsername() {
        val errors = mutableListOf<ValidationErrorModel>()
        
        if (username.value.isEmpty()) {
            errors.add(
                ValidationErrorModel(
                    code = "required",
                    messageKey = "validation.required",
                    metadata = emptyMap()
                )
            )
        }
        
        if (username.value.length < 3 && username.value.isNotEmpty()) {
            errors.add(
                ValidationErrorModel(
                    code = "minLength",
                    messageKey = "validation.minLength",
                    metadata = mapOf(
                        "minLength" to "3",
                        "currentLength" to username.value.length.toString()
                    )
                )
            )
        }
        
        if (username.value.length > 20) {
            errors.add(
                ValidationErrorModel(
                    code = "maxLength",
                    messageKey = "validation.maxLength",
                    metadata = mapOf(
                        "maxLength" to "20",
                        "currentLength" to username.value.length.toString()
                    )
                )
            )
        }
        
        val validPattern = Regex("^[a-zA-Z0-9_-]+$")
        if (!validPattern.matches(username.value) && username.value.isNotEmpty()) {
            errors.add(
                ValidationErrorModel(
                    code = "username.invalid",
                    messageKey = "validation.username.alphanumericOnly",
                    metadata = emptyMap()
                )
            )
        }
        
        val result = if (errors.isEmpty()) {
            ValidationResultModel(isValid = true, errors = null)
        } else {
            ValidationResultModel(isValid = false, errors = errors)
        }
        
        isUsernameValid.value = result.isValid
        usernameError.value = result.errors?.firstOrNull()
    }
    
    private fun validatePassword() {
        val errors = mutableListOf<ValidationErrorModel>()
        
        if (password.value.isEmpty()) {
            errors.add(
                ValidationErrorModel(
                    code = "required",
                    messageKey = "validation.required",
                    metadata = emptyMap()
                )
            )
        }
        
        if (password.value.length < 8 && password.value.isNotEmpty()) {
            errors.add(
                ValidationErrorModel(
                    code = "minLength",
                    messageKey = "validation.minLength",
                    metadata = mapOf(
                        "minLength" to "8",
                        "currentLength" to password.value.length.toString()
                    )
                )
            )
        }
        
        if (password.value.length > 128) {
            errors.add(
                ValidationErrorModel(
                    code = "maxLength",
                    messageKey = "validation.maxLength",
                    metadata = emptyMap()
                )
            )
        }
        
        val hasUpperAndLower = 
            password.value.any { it.isUpperCase() } && 
            password.value.any { it.isLowerCase() }
        val hasDigit = password.value.any { it.isDigit() }
        
        if ((!hasUpperAndLower || !hasDigit) && password.value.isNotEmpty()) {
            errors.add(
                ValidationErrorModel(
                    code = "password.weak",
                    messageKey = "validation.password.requiresUpperLowerNumber",
                    metadata = emptyMap()
                )
            )
        }
        
        val result = if (errors.isEmpty()) {
            ValidationResultModel(isValid = true, errors = null)
        } else {
            ValidationResultModel(isValid = false, errors = errors)
        }
        
        isPasswordValid.value = result.isValid
        passwordError.value = result.errors?.firstOrNull()
    }
    
    private fun validateConfirmPassword() {
        val errors = mutableListOf<ValidationErrorModel>()
        
        if (confirmPassword.value.isEmpty()) {
            errors.add(
                ValidationErrorModel(
                    code = "required",
                    messageKey = "validation.required",
                    metadata = emptyMap()
                )
            )
        }
        
        if (confirmPassword.value != password.value && confirmPassword.value.isNotEmpty()) {
            errors.add(
                ValidationErrorModel(
                    code = "equals",
                    messageKey = "validation.equals",
                    metadata = emptyMap()
                )
            )
        }
        
        val result = if (errors.isEmpty()) {
            ValidationResultModel(isValid = true, errors = null)
        } else {
            ValidationResultModel(isValid = false, errors = errors)
        }
        
        isConfirmPasswordValid.value = result.isValid
        confirmPasswordError.value = result.errors?.firstOrNull()
    }
    
    private fun validateTextField(value: String, min: Int, max: Int): ValidationResultModel {
        val errors = mutableListOf<ValidationErrorModel>()
        
        if (value.isEmpty()) {
            errors.add(
                ValidationErrorModel(
                    code = "required",
                    messageKey = "validation.required",
                    metadata = emptyMap()
                )
            )
        }
        
        if (value.length < min && value.isNotEmpty()) {
            errors.add(
                ValidationErrorModel(
                    code = "minLength",
                    messageKey = "validation.minLength",
                    metadata = mapOf(
                        "minLength" to min.toString(),
                        "currentLength" to value.length.toString()
                    )
                )
            )
        }
        
        if (value.length > max) {
            errors.add(
                ValidationErrorModel(
                    code = "maxLength",
                    messageKey = "validation.maxLength",
                    metadata = mapOf(
                        "maxLength" to max.toString(),
                        "currentLength" to value.length.toString()
                    )
                )
            )
        }
        
        return if (errors.isEmpty()) {
            ValidationResultModel(isValid = true, errors = null)
        } else {
            ValidationResultModel(isValid = false, errors = errors)
        }
    }
    
    val isFormValid: Boolean
        get() = isFirstNameValid.value && isLastNameValid.value && 
                isEmailValid.value && isUsernameValid.value && 
                isPasswordValid.value && isConfirmPasswordValid.value
    
    val completionPercentage: Float
        get() {
            val validFields = listOf(
                isFirstNameValid.value, isLastNameValid.value,
                isEmailValid.value, isUsernameValid.value,
                isPasswordValid.value, isConfirmPasswordValid.value
            ).count { it }
            
            return validFields / 6f
        }
}

// Form Field Component
@Composable
fun SignUpFormField(
    label: String,
    icon: androidx.compose.material.icons.filled.Icon,
    placeholder: String,
    value: String,
    onValueChange: (String) -> Unit,
    error: ValidationErrorModel?,
    isValid: Boolean,
    isPassword: Boolean = false
) {
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .padding(bottom = 8.dp),
        verticalArrangement = Arrangement.spacedBy(4.dp)
    ) {
        Row(
            modifier = Modifier.fillMaxWidth(),
            verticalAlignment = Alignment.CenterVertically
        ) {
            Text(label, style = MaterialTheme.typography.labelLarge)
            Spacer(modifier = Modifier.weight(1f))
            if (isValid && value.isNotEmpty()) {
                Icon(
                    Icons.Default.CheckCircle,
                    contentDescription = null,
                    tint = Color.Green,
                    modifier = Modifier.size(16.dp)
                )
            }
        }
        
        OutlinedTextField(
            value = value,
            onValueChange = onValueChange,
            label = { Text(placeholder) },
            modifier = Modifier.fillMaxWidth(),
            singleLine = true
        )
        
        if (error != null && value.isNotEmpty()) {
            Card(
                modifier = Modifier.fillMaxWidth(),
                colors = CardDefaults.cardColors(
                    containerColor = Color(0xFFFFEBEE)
                )
            ) {
                Column(
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(8.dp)
                ) {
                    Text(
                        error.messageKey,
                        style = MaterialTheme.typography.bodySmall,
                        color = Color(0xFFC62828)
                    )
                    if (error.metadata.isNotEmpty()) {
                        Text(
                            error.metadata.map { "${it.key}: ${it.value}" }.joinToString(" • "),
                            style = MaterialTheme.typography.bodySmall,
                            color = Color.Gray
                        )
                    }
                }
            }
        }
    }
}

// Progress Indicator
@Composable
fun FormProgressIndicator(percentage: Float) {
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .padding(vertical = 12.dp),
        verticalArrangement = Arrangement.spacedBy(8.dp)
    ) {
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.SpaceBetween
        ) {
            Text(
                "Form Completion",
                style = MaterialTheme.typography.labelSmall
            )
            Text(
                "${(percentage * 100).toInt()}%",
                style = MaterialTheme.typography.labelSmall,
                color = MaterialTheme.colorScheme.primary
            )
        }
        
        LinearProgressIndicator(
            progress = percentage,
            modifier = Modifier
                .fillMaxWidth()
                .height(8.dp),
            color = MaterialTheme.colorScheme.primary
        )
    }
}

// Main Sign-Up Form Screen
@Composable
fun SignUpFormDemoScreen(
    viewModel: SignUpFormViewModel = viewModel()
) {
    var showSuccess by remember { mutableStateOf(false) }
    
    Column(
        modifier = Modifier
            .fillMaxSize()
            .verticalScroll(rememberScrollState())
            .padding(20.dp),
        verticalArrangement = Arrangement.spacedBy(16.dp)
    ) {
        // Header
        Text(
            "Create Account",
            style = MaterialTheme.typography.headlineSmall
        )
        
        Text(
            "Sign up for UniCore Validation Demo",
            style = MaterialTheme.typography.bodySmall,
            color = MaterialTheme.colorScheme.onSurfaceVariant
        )
        
        // Progress
        FormProgressIndicator(viewModel.completionPercentage)
        
        Divider()
        
        // Personal Information Section
        Text(
            "Personal Information",
            style = MaterialTheme.typography.labelLarge
        )
        
        SignUpFormField(
            label = "First Name",
            icon = Icons.Default.Person,
            placeholder = "John",
            value = viewModel.firstName.value,
            onValueChange = { viewModel.updateFirstName(it) },
            error = viewModel.firstNameError.value,
            isValid = viewModel.isFirstNameValid.value
        )
        
        SignUpFormField(
            label = "Last Name",
            icon = Icons.Default.Person,
            placeholder = "Doe",
            value = viewModel.lastName.value,
            onValueChange = { viewModel.updateLastName(it) },
            error = viewModel.lastNameError.value,
            isValid = viewModel.isLastNameValid.value
        )
        
        Divider()
        
        // Account Information Section
        Text(
            "Account Information",
            style = MaterialTheme.typography.labelLarge
        )
        
        SignUpFormField(
            label = "Email",
            icon = Icons.Default.Email,
            placeholder = "user@example.com",
            value = viewModel.email.value,
            onValueChange = { viewModel.updateEmail(it) },
            error = viewModel.emailError.value,
            isValid = viewModel.isEmailValid.value
        )
        
        SignUpFormField(
            label = "Username",
            icon = Icons.Default.Person,
            placeholder = "john_doe",
            value = viewModel.username.value,
            onValueChange = { viewModel.updateUsername(it) },
            error = viewModel.usernameError.value,
            isValid = viewModel.isUsernameValid.value
        )
        
        Divider()
        
        // Security Section
        Text(
            "Security",
            style = MaterialTheme.typography.labelLarge
        )
        
        SignUpFormField(
            label = "Password",
            icon = Icons.Default.Lock,
            placeholder = "••••••••",
            value = viewModel.password.value,
            onValueChange = { viewModel.updatePassword(it) },
            error = viewModel.passwordError.value,
            isValid = viewModel.isPasswordValid.value,
            isPassword = true
        )
        
        SignUpFormField(
            label = "Confirm Password",
            icon = Icons.Default.Lock,
            placeholder = "••••••••",
            value = viewModel.confirmPassword.value,
            onValueChange = { viewModel.updateConfirmPassword(it) },
            error = viewModel.confirmPasswordError.value,
            isValid = viewModel.isConfirmPasswordValid.value,
            isPassword = true
        )
        
        Spacer(modifier = Modifier.height(12.dp))
        
        // Action Buttons
        Button(
            onClick = { showSuccess = true },
            modifier = Modifier
                .fillMaxWidth()
                .height(48.dp),
            enabled = viewModel.isFormValid
        ) {
            Text("Create Account")
        }
        
        if (showSuccess) {
            AlertDialog(
                onDismissRequest = { showSuccess = false },
                title = { Text("Success") },
                text = { Text("Account created successfully!") },
                confirmButton = {
                    Button(onClick = { showSuccess = false }) {
                        Text("OK")
                    }
                }
            )
        }
        
        Spacer(modifier = Modifier.height(20.dp))
    }
}

File: Examples/iOS/SignUpFormDemo.swift
import SwiftUI
import UniCoreValidation

/// Comprehensive sign-up form with multiple field validation
@Observable
final class SignUpFormViewModel {
    var firstName: String = "" {
        didSet { validateFirstName() }
    }
    
    var lastName: String = "" {
        didSet { validateLastName() }
    }
    
    var email: String = "" {
        didSet { validateEmail() }
    }
    
    var username: String = "" {
        didSet { validateUsername() }
    }
    
    var password: String = "" {
        didSet { validatePassword() }
    }
    
    var confirmPassword: String = "" {
        didSet { validateConfirmPassword() }
    }
    
    // MARK: - Validation State
    var firstNameError: ValidationError?
    var lastNameError: ValidationError?
    var emailError: ValidationError?
    var usernameError: ValidationError?
    var passwordError: ValidationError?
    var confirmPasswordError: ValidationError?
    
    var isFirstNameValid = false
    var isLastNameValid = false
    var isEmailValid = false
    var isUsernameValid = false
    var isPasswordValid = false
    var isConfirmPasswordValid = false
    
    // MARK: - Validators
    @ObservationIgnored
    private let firstNameValidator = Validator(rules: [
        RequiredRule(),
        MinLengthRule(minLength: 2),
        MaxLengthRule(maxLength: 50)
    ])
    
    @ObservationIgnored
    private let lastNameValidator = Validator(rules: [
        RequiredRule(),
        MinLengthRule(minLength: 2),
        MaxLengthRule(maxLength: 50)
    ])
    
    @ObservationIgnored
    private let emailValidator = Validator(rules: [
        RequiredRule(),
        EmailRule()
    ])
    
    @ObservationIgnored
    private lazy var usernameValidator = Validator(rules: [
        RequiredRule(),
        MinLengthRule(minLength: 3),
        MaxLengthRule(maxLength: 20),
        RegexRule(
            pattern: "^[a-zA-Z0-9_-]+$",
            code: "username.invalid",
            messageKey: "validation.username.alphanumericOnly"
        )
    ])
    
    @ObservationIgnored
    private let passwordValidator = Validator(rules: [
        RequiredRule(),
        MinLengthRule(minLength: 8),
        MaxLengthRule(maxLength: 128),
        RegexRule(
            pattern: "^(?=.*[a-z])(?=.*[A-Z])(?=.*\\d).*$",
            code: "password.weak",
            messageKey: "validation.password.requiresUpperLowerNumber"
        )
    ])
    
    // MARK: - Validation Methods
    private func validateFirstName() {
        let result = firstNameValidator.validate(firstName)
        isFirstNameValid = result.isValid
        firstNameError = result.errors?.first
    }
    
    private func validateLastName() {
        let result = lastNameValidator.validate(lastName)
        isLastNameValid = result.isValid
        lastNameError = result.errors?.first
    }
    
    private func validateEmail() {
        let result = emailValidator.validate(email)
        isEmailValid = result.isValid
        emailError = result.errors?.first
    }
    
    private func validateUsername() {
        let result = usernameValidator.validate(username)
        isUsernameValid = result.isValid
        usernameError = result.errors?.first
    }
    
    private func validatePassword() {
        let result = passwordValidator.validate(password)
        isPasswordValid = result.isValid
        passwordError = result.errors?.first
    }
    
    private func validateConfirmPassword() {
        let matchValidator = Validator(rules: [
            RequiredRule(),
            EqualsRule(expectedValue: password)
        ])
        
        let result = matchValidator.validate(confirmPassword)
        isConfirmPasswordValid = result.isValid
        confirmPasswordError = result.errors?.first
    }
    
    var isFormValid: Bool {
        isFirstNameValid && isLastNameValid && isEmailValid &&
        isUsernameValid && isPasswordValid && isConfirmPasswordValid
    }
    
    var completionPercentage: Double {
        let validFields = [
            isFirstNameValid, isLastNameValid, isEmailValid,
            isUsernameValid, isPasswordValid, isConfirmPasswordValid
        ].filter { $0 }.count
        
        return Double(validFields) / 6.0
    }
}

// MARK: - Form Field Component
struct FormField: View {
    let label: String
    let icon: String
    let placeholder: String
    @Binding var text: String
    let error: ValidationError?
    let isValid: Bool
    let isSecure: Bool
    @State private var showSecure = true
    
    var body: some View {
        VStack(alignment: .leading, spacing: 6) {
            HStack {
                Label(label, systemImage: icon)
                    .font(.subheadline)
                    .fontWeight(.semibold)
                
                Spacer()
                
                if isValid && !text.isEmpty {
                    Image(systemName: "checkmark.circle.fill")
                        .foregroundColor(.green)
                        .font(.caption)
                }
            }
            
            HStack {
                if isSecure && showSecure {
                    SecureField(placeholder, text: $text)
                } else {
                    TextField(placeholder, text: $text)
                        .textInputAutocapitalization(.never)
                }
                
                if isSecure {
                    Button(action: { showSecure.toggle() }) {
                        Image(systemName: showSecure ? "eye.slash" : "eye")
                            .foregroundColor(.blue)
                            .font(.caption)
                    }
                }
            }
            .padding(10)
            .background(Color(.systemGray6))
            .cornerRadius(6)
            
            if let error = error, !text.isEmpty {
                HStack(spacing: 6) {
                    Image(systemName: "exclamationmark.circle.fill")
                        .foregroundColor(.red)
                        .font(.caption)
                    
                    VStack(alignment: .leading, spacing: 2) {
                        Text(error.messageKey)
                            .font(.caption2)
                            .fontWeight(.semibold)
                        
                        if let details = error.metadata.map({ "\($0.key): \($0.value)" }).joined(separator: " • "), !details.isEmpty {
                            Text(details)
                                .font(.caption2)
                                .foregroundColor(.gray)
                        }
                    }
                }
                .padding(8)
                .background(Color.red.opacity(0.1))
                .cornerRadius(6)
            }
        }
    }
}

// MARK: - Progress Indicator
struct FormProgressView: View {
    let percentage: Double
    
    var body: some View {
        VStack(alignment: .leading, spacing: 8) {
            HStack {
                Text("Form Completion")
                    .font(.caption)
                    .fontWeight(.semibold)
                
                Spacer()
                
                Text("\(Int(percentage * 100))%")
                    .font(.caption)
                    .fontWeight(.semibold)
                    .foregroundColor(.blue)
            }
            
            GeometryReader { geometry in
                ZStack(alignment: .leading) {
                    Color(.systemGray5)
                    
                    Color.blue
                        .frame(width: geometry.size.width * percentage)
                }
            }
            .frame(height: 8)
            .cornerRadius(4)
        }
    }
}

// MARK: - Main Sign-Up Form View
struct SignUpFormDemo: View {
    @State private var viewModel = SignUpFormViewModel()
    @State private var showSuccessAlert = false
    
    var body: some View {
        NavigationStack {
            ScrollView {
                VStack(spacing: 20) {
                    // Header
                    VStack(spacing: 8) {
                        Text("Create Account")
                            .font(.title2)
                            .fontWeight(.bold)
                        
                        Text("Sign up for UniCore Validation Demo")
                            .font(.caption)
                            .foregroundColor(.secondary)
                    }
                    .frame(maxWidth: .infinity, alignment: .leading)
                    
                    // Progress
                    FormProgressView(percentage: viewModel.completionPercentage)
                    
                    // Personal Info Section
                    VStack(spacing: 14) {
                        Text("Personal Information")
                            .font(.headline)
                            .frame(maxWidth: .infinity, alignment: .leading)
                        
                        FormField(
                            label: "First Name",
                            icon: "person",
                            placeholder: "John",
                            text: $viewModel.firstName,
                            error: viewModel.firstNameError,
                            isValid: viewModel.isFirstNameValid,
                            isSecure: false
                        )
                        
                        FormField(
                            label: "Last Name",
                            icon: "person.fill",
                            placeholder: "Doe",
                            text: $viewModel.lastName,
                            error: viewModel.lastNameError,
                            isValid: viewModel.isLastNameValid,
                            isSecure: false
                        )
                    }
                    .padding(14)
                    .background(Color(.systemGray6))
                    .cornerRadius(10)
                    
                    // Account Info Section
                    VStack(spacing: 14) {
                        Text("Account Information")
                            .font(.headline)
                            .frame(maxWidth: .infinity, alignment: .leading)
                        
                        FormField(
                            label: "Email",
                            icon: "envelope",
                            placeholder: "user@example.com",
                            text: $viewModel.email,
                            error: viewModel.emailError,
                            isValid: viewModel.isEmailValid,
                            isSecure: false
                        )
                        
                        FormField(
                            label: "Username",
                            icon: "at",
                            placeholder: "john_doe",
                            text: $viewModel.username,
                            error: viewModel.usernameError,
                            isValid: viewModel.isUsernameValid,
                            isSecure: false
                        )
                    }
                    .padding(14)
                    .background(Color(.systemGray6))
                    .cornerRadius(10)
                    
                    // Security Section
                    VStack(spacing: 14) {
                        Text("Security")
                            .font(.headline)
                            .frame(maxWidth: .infinity, alignment: .leading)
                        
                        FormField(
                            label: "Password",
                            icon: "lock",
                            placeholder: "••••••••",
                            text: $viewModel.password,
                            error: viewModel.passwordError,
                            isValid: viewModel.isPasswordValid,
                            isSecure: true
                        )
                        
                        FormField(
                            label: "Confirm Password",
                            icon: "lock.fill",
                            placeholder: "••••••••",
                            text: $viewModel.confirmPassword,
                            error: viewModel.confirmPasswordError,
                            isValid: viewModel.isConfirmPasswordValid,
                            isSecure: true
                        )
                    }
                    .padding(14)
                    .background(Color(.systemGray6))
                    .cornerRadius(10)
                    
                    // Actions
                    VStack(spacing: 10) {
                        Button(action: { showSuccessAlert = true }) {
                            Text("Create Account")
                                .frame(maxWidth: .infinity)
                                .padding(12)
                                .background(viewModel.isFormValid ? Color.blue : Color.gray)
                                .foregroundColor(.white)
                                .cornerRadius(8)
                                .fontWeight(.semibold)
                        }
                        .disabled(!viewModel.isFormValid)
                        
                        NavigationLink(destination: Text("Account created!")) {
                            Text("Already have an account?")
                                .frame(maxWidth: .infinity)
                                .padding(12)
                                .foregroundColor(.blue)
                        }
                    }
                    
                    Spacer()
                }
                .padding(20)
            }
            .navigationTitle("Sign Up")
            .navigationBarTitleDisplayMode(.inline)
            .alert("Success", isPresented: $showSuccessAlert) {
                Button("OK") { }
            } message: {
                Text("Account created successfully!")
            }
        }
    }
}

#Preview {
    SignUpFormDemo()
}
